src/com/projectkorra/projectkorra/region/RegionProtection.java

```diff
@@ -176,14 +172,20 @@ private static boolean checkAll(Player player, Location location, CoreAbility ab
      */
     public static void startCleanCacheTask(double period) {
         Bukkit.getScheduler().runTaskTimer(ProjectKorra.plugin, () -> {
-            for (final Map<Block, BlockCacheElement> map : BLOCK_CACHE.values()) {
+            final long currentTime = System.currentTimeMillis();
+            for (final String playerName : BLOCK_CACHE.keySet()) {
+                final Map<Block, BlockCacheElement> map = BLOCK_CACHE.get(playerName);
                 for (final Block key : map.keySet()) {
                     final BlockCacheElement value = map.get(key);
 
-                    if (System.currentTimeMillis() - value.getTime() > period) {
+                    if (currentTime - value.getTime() > period) {
                         map.remove(key);
                     }
                 }
+                if (map.size() == 0) {
```

Comment:
Without cleaning up empty maps from BLOCK_CACHE the server will accumulate empty maps for every user that logs in and bends, without ever dereferencing them so that they can be garbage collected.